README.md

```diff
@@ -26,7 +38,213 @@ required for testing citus.
+* Variables
+* Resources
+* Outputs
+
+Parameters can be configured from the parameters file. Variables are constants. Resources have all of the resource definitions such as VMs, network security groups. Outputs can be useful for displaying a connection string.
+
+When creating resources, we can specify the order so that if a resource depends on some other resource, it wont be created before the dependant is created. We can also specify how many instances of a resource to create with a `copy` command.
+
+The first virtual machine with index 0 is treated as a coordinator. When all the virtual machines are ready, a custom script is installed to do initialization in vms. The initailization script is retrieved from the github with a url.
+
+The initialization script also finds the private ip addresses of workers and puts them to the coordinator. The way this is done is with a storage account resource. This storage account resource is created within the template itself and all the vms upload their private ip addresses to the storage. After all are uploaded the coordinator downloads all the private ip addresses from the storage account and puts it to `worker-instances` file, which is then used when creating a citus cluster.
+
+We have a special security group which blocks ssh traffic. The rule's priority is 103 and 100, 101, 102 are also taken by this security group. The workaround to connect to the coordinator is to remove the rule 103, and connect right after it. The rule comes back in 2-3 mins, so you should be fast here. There is a script called `delete_security_rule.sh`, which deletes that rule for you.
+
+Before starting the process you should set the environment variable `RESOURCE_GROUP_NAME`, which is used in all scripts.
+
+```bash
+export RESOURCE_GROUP_NAME=talhatest_resource_group
+```
+
+You should use a single session because the exported variable is only available in the current session and its children sessions. You should start `ssh-agent` and add your key with `ssh-add`.
+
+Even if you want to use the defaults, you still need to put your public key to parameters. This public key will be put to the virtual machines so that you can ssh to them.
+
+To simplify this process, there is a script called `create-cluster.sh`, which:
+
+* creates a resource group from the environment variable `RESOURCE_GROUP_NAME`
+* creates a cluster with the `azuredeploy.json` template in the resource group
+* prints the connection string to ssh
+
+then you should run:
+
+```bash
+./delete_security_rule.sh
+ssh -A pguser@<public ip of coordinator>
+```
+
+After you are done with testing, you can delete the resource group with:
+
+```bash
+./delete_resource_group.sh
+```
+
+Currently each pgbench test is run for 300 seconds, and as we have many pgbench tests it might take a while to run all the tests. So when testing a change, it is better to change the tests times to something short such as 5 seconds. The time can be changed with the -T parameter:
```

Comment:
This seems like it does not belong in the Azure section.

Reply:
I will update this to general test instead of just pgbench, because other tests have timing parameter too. Pgbench is just one of them.